rts/motoko-rts/src/gc/incremental/partitioned_heap.rs

```diff
@@ -515,26 +701,40 @@ impl PartitionedHeap {
         }
     }
 
-    fn rank_partitions_by_garbage(&self) -> [usize; MAX_PARTITIONS] {
-        let mut ranked_partitions: [usize; MAX_PARTITIONS] = from_fn(|index| index);
```

Comment:
I'm curious why this function even works? Doesn't it return a pointer to something stack allocated? Or is copying the array?

Reply:
The caller stack-allocates the space for the return value and only passes a pointer to that location. This allows bulk return values without copying. (This is also done in C and in Oberon :))